gemrb/core/GUI/TextSystem/GemMarkup.cpp

```diff
@@ -25,11 +25,11 @@ namespace GemRB {
 static Color ParseColor(const String& colorString)
 {
 	Color color = ColorWhite;
-	uint8_t values[4] = {0, 0, 0, 0};
+	uint8_t values[4] = { 0, 0, 0, 0xff };
 
 	// That's a short way to interpret 4 groups of %02hhx,
 	// since `swscanf` does not work on our string type unless converting first
-	for (uint8_t i = 0; i < 4; ++i) {
+	for (uint8_t i = 0; i < colorString.size() / 2; ++i) {
```

Comment:
I don't understand why `/ 2` is here. shouldn't `colorString.size()` be the number of nibbles represented in the string? I guess not, but I dont know what `colorString` looks like, is it padded with `0x00` for some reason? Maybe we should fix that.

Reply:
![Screenshot 2023-11-29 224227](https://github.com/gemrb/gemrb/assets/119408536/6cd05d8c-05ba-4396-b125-bbf902ef621f)

Reply:
right, so it is just the number of nibbles. we should iterate them all we just have some funky offset voodoo that we can probably get rid of instead.